service/test/agama/storage/manager_test.rb

```diff
@@ -315,17 +330,30 @@
     let(:devicegraph) { "staging-plain-partitions.yaml" }
 
     it "copy needed files, installs the bootloader, sets up the snapshots, " \
-       "copy logs, and umounts the file systems" do
+       "copy logs, runs the post-installation scripts, and umounts the file systems" do
       expect(security).to receive(:write)
       expect(copy_files).to receive(:run)
       expect(bootloader_finish).to receive(:write)
       expect(Yast::WFM).to receive(:CallFunction).with("storage_finish", ["Write"])
       expect(Yast::WFM).to receive(:CallFunction).with("snapshots_finish", ["Write"])
+      expect(scripts_client).to receive(:run).with("post")
       expect(Yast::WFM).to receive(:CallFunction).with("copy_logs_finish", ["Write"])
       expect(Yast::WFM).to receive(:CallFunction).with("umount_finish", ["Write"])
       storage.finish
     end
 
+    context "when scripts artifacts exist" do
+      before do
+        FileUtils.mkdir_p(scripts_dir)
+        FileUtils.touch(File.join(scripts_dir, "test.sh"))
+      end
+
+      it "copies the artifacts to the installed system" do
+        storage.finish
+        expect(File).to exist(File.join(tmp_dir, "mnt", "var", "log", "agama", "scripts"))
```

Comment:
I'm not sure if the test should really touch the testing system (create and copy files). I do not see much advantage over just to expect to call `FileUtils.cp_r`.

Reply:
I prefer this (that's why I use a temporary directory) instead of mocking all the calls, to be honest.

Reply:
Fine for me, it's just a matter of preferences.

Reply:
I have changed the code to store those assets under `/var/log/agama-installation`.